### ISSUE TITLE
This is the best bot I have ever seen in GitHub. Thank you for your time and effort.

### ISSUE THREAD

**Issue by cyborg2006** [mustafadalga/Instagram-Bot#5]
Hi Mustafa, 
First of all your code is very clear, understandable and perfectly working. I hope you will update whenever IG change the HTML codes. Thank you for your time and appreciate to share this code with the community.

I have some comments and suggestions for your review:

1) We should see all the accounts in CSV or TXT file which are not following us even we followed them. You did a great job to found them before unfollow, but Bot is continuously unfollowing them without any schedule. This will cause very quick ban by IG. You should consider to put a adjustable schedule to start unfollowing option. 
2) We should have a list of accounts which we followed and which users followed us in CSV or TXT file for the accounts. 
3) You should find the accounts which we followed but not followed us and print it to the screen one by one asking below options:
   - just unfollow
   - add to blacklist
   - add to whitelist
We shouldn't unfollow any account which is in whitelist. 
4) Bot is not working if the account has lots of followers like 56K. The letter "K" cause a problem to start the code. 
 
Please let us know what you think.

**Comment by mustafadalga** [comment]
Hello @cyborg2006,
Thank you for your feedback.

**For 1 suggestion;**
It may be unnecessary to use csv or txt files for this suggestion.
Time intervals are specified for each process in config.json file. You can change the time intervals in config.json.
If you want to change the time interval of which action or method, find the name of the method in config.json and change the times.So you can solution the problem.

**For example: Bulk unfollow feature**
You can see how long each user waits after unfollowing:

**instagram.py**
https://github.com/mustafadalga/Instagram-Bot/blob/88396d3a2e94a6b274c228fb0912d720e6674d30/instagram.py#L295

Wait time values ​​are set in config.json like this:

**config.py**
https://github.com/mustafadalga/Instagram-Bot/blob/88396d3a2e94a6b274c228fb0912d720e6674d30/config.json#L1528


**Summary:** The processing waiting time of each method can be changed in config.json.Thus, the problem of blocking by instagram is solved.

**For 2 suggestion:**
we can save followers and following as csv or txt.

**For 3 suggestion:**
If we ask each user individually, the process will be very long. Can you give a little more detail about this feature? I want to understand what you mean.

and **for item 4**  an error occurred while using which feature ? I will examine the issue you reported.

**Comment by cyborg2006** [comment]
Hi,

Thank you for your time and answers. I will do that. Please see below detail for item 4. 
I reached one of my account tried to use option 6, but I received below response. 

[quote]
 Enter an option number >> 6

Selected Option >>> Unfollow Users Who Don't Follow
[*] Enter 'menu' command to return main menu

[*] Adding followers to the list is started.
[*] Browser is redirected to your profile...
[-] An error occurred while selecting followers: invalid literal for int() with base 10: '37k'

[unquote]

for Item 3:
For instance, you found a account name which we followed by not followed us name "xxxxx".
You should show the name "xxxxx" on screen asking 1) unfollow 2) Add white list. 3) Skip. After our answer bot will continue to show next account, and other, .... so on...
We have to keep some accounts followed even they haven't followed us back. We need a white list to save some accounts never unfollowed by us even accidentally. This is long procedure, but if we have this option as well. I am sure lots of people wants to use it.

**Comment by mustafadalga** [comment]
@cyborg2006 
I took note of your suggestions.

I solved the problem in item 4.Would you try the same action again?

**Comment by cyborg2006** [comment]
Now, it is working. Thanks for your update.

**Comment by mustafadalga** [comment]
You're welcome 🙂 . I will work on more features and options for the next version.

**Comment by Arakiss** [comment]
I agree with the OP, very good job with this bot. Thank you very much. I have a small suggestion. The possibility of excluding some users that do not follow us. I only missed that option.

Thanks!
